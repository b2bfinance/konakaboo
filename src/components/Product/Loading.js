import React from 'react';
import styled, { css } from 'styled-components';
import { queries } from '../../utils/media';

export const Wrapper = styled.div`
  ${props =>
    props.loading &&
    css`
      opacity: 0.75;
    `};
`;

export const MaskItem = styled.div`
  background-color: ${props => props.theme.productMaskBackground};
  height: 35rem;
  margin-bottom: 1.5rem;

  ${queries.desktop`
    height: 9rem;
  `};
`;

export const Mask = ({ initiated, theme }) =>
  !initiated && (
    <React.Fragment>
      <MaskItem theme={theme} />
      <MaskItem theme={theme} />
      <MaskItem theme={theme} />
    </React.Fragment>
  );

export default ({ loading, children }) => (
  <Wrapper loading={loading}>{children}</Wrapper>
);
