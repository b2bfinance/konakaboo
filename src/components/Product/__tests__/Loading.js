import React from 'react';
import { render } from 'enzyme';
import Loading, { Mask } from '../Loading';

describe('Loading', () => {
  test('renders correctly when not loading', () => {
    expect(render(<Loading>foo</Loading>)).toMatchSnapshot();
  });

  test('renders correctly when no loading', () => {
    expect(render(<Loading loading>foo</Loading>)).toMatchSnapshot();
  });
});

describe('Mask', () => {
  test('renders correctly when not initiated', () => {
    expect(render(<Mask theme={stubData.theme} />)).toMatchSnapshot();
  });

  test('renders correctly when no initiated', () => {
    expect(render(<Mask initiated theme={stubData.theme} />)).toMatchSnapshot();
  });
});
