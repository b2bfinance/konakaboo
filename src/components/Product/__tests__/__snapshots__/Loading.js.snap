// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly when no loading 1`] = `
.c0 {
  opacity: 0.75;
}

<div
  class="c0"
>
  foo
</div>
`;

exports[`Loading renders correctly when not loading 1`] = `
<div
  class=""
>
  foo
</div>
`;

exports[`Mask renders correctly when no initiated 1`] = `null`;

exports[`Mask renders correctly when not initiated 1`] = `
Array [
  .c0 {
  background-color: #eceff1;
  height: 35rem;
  margin-bottom: 1.5rem;
}

@media (min-width:62em) {
  .c0 {
    height: 9rem;
  }
}

<div
    class="c0"
  />,
  .c0 {
  background-color: #eceff1;
  height: 35rem;
  margin-bottom: 1.5rem;
}

@media (min-width:62em) {
  .c0 {
    height: 9rem;
  }
}

<div
    class="c0"
  />,
  .c0 {
  background-color: #eceff1;
  height: 35rem;
  margin-bottom: 1.5rem;
}

@media (min-width:62em) {
  .c0 {
    height: 9rem;
  }
}

<div
    class="c0"
  />,
]
`;
